lnwallet/wallet.go: clarify anchor chan error str

xanoni opened the pull request:
See https://github.com/lightningnetwork/lnd/issues/5523

Improve 'ErrReservedValueInvalidated' error string to explain that the
error is triggered by a transaction that would deplete funds reserved for
potential future anchor channel closings (via CPFP)

Add hint that further details can be found in the debug log

Update strings in 'lntest/itest/log_error_whitelist.txt' correspondingly

guggero commented on lnwallet/wallet.go:
```diff
@@ -59,7 +59,8 @@ var (
 	// to keep around to fee bump our open anchor channels.
 	ErrReservedValueInvalidated = errors.New("reserved wallet balance " +
-		"invalidated")
+		"invalidated: transaction would not leave sufficient funds to " +
```
nit: how about `would leave insufficient reserve for fee bumping ...` instead of `not leave sufficient funds to fee bump ...`?

guggero reviewed (commented):
Thanks for the fix. The new error message is definitely more understandable. Added suggestion of my own, hope you don't mind.

All PRs need an entry in the release notes (this one should go into 0.14.0) too.

xanoni commented:
> Thanks for the fix. The new error message is definitely more understandable. Added suggestion of my own, hope you don't mind.

Yeah sure it was late and that simple sentence took me far too long to construct, but I also noticed that it was still a bit bumpy, haha.

Saw the failing release notes test but assumed this was too insignificant to be mentioned. It's probably all explained in the contributor guidelines.

Will update all that later and read these guidelines.

xanoni commented:
Quick update, I remember that you mentioned that both sides of the channel need the 10k reserve. I need to look at the code again to understand if it would raise the same error message if the other node is lacking funds, and if yes, include that hint in the error string.

carlaKC reviewed (approved):
Great improvement to this error! 

Just need to add this to the 0.14 release notes, no PR is too small! :)

xanoni commented:
> Just need to add this to the 0.14 release notes, no PR is too small! :)

Thank you! Life's been busy but I'll definitely get to it this weekend. 

**TODO:** 
- Understand what message is thrown when the other node doesn't have sufficient funds, and if that message needs to be clarified as well
- Add release notes

xanoni commented:
I think this should be ready now ... I updated the strings as suggested by @guggero and added the PR to the release notes.

I also checked whether the same error is thrown when the counterparty node doesn't have sufficient funds, but couldn't see that that's the case. So either only the local node is checked, or the balance of the other node is checked somewhere else and a different error is thrown.

xanoni commented:
Thank you!
